source/BannerlordCoopInstaller/Installer.cs

```diff
@@ -0,0 +1,264 @@
+                        File.WriteAllText($@"{this.BannerlordPath.Text}\Modules\Coop{resourceFile.InstallPath}\{fileName}", resource.Value.ToString());
+                    }
+                    
+                }
+                else
+                {
+                    string fileName = resource.Key + ".dll";
+                    this.Status.Text = "Copying " + fileName + "...";
+                    File.WriteAllBytes($@"{this.BannerlordPath.Text}\Modules\Coop\bin\Win64_Shipping_Client\{fileName}" , (byte[])resource.Value);
+                }
+
+            }
+        }
+
+        /// <summary>
+        /// Disable controls, check if a version of the mod is installed; ask the user to delete it. Install the mod and change install button to finish
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void Install_Click(object sender, EventArgs e)
+        {
+            this.BannerlordPath.Enabled = false;
+            this.ShortcutCheckbox.Enabled = false;
+            this.PathBrowse.Enabled = false;
+            
+            // version exists, ask to delete it
+            if (Directory.Exists($@"{this.BannerlordPath.Text}\Modules\Coop"))
+            {
+                DialogResult dialogResult = MessageBox.Show("An older version exists and must be removed. Are you sure?", "Older Version Detected", MessageBoxButtons.YesNo);
+                if (dialogResult == DialogResult.Yes)
+                {
+                    try
+                    {
+                        //recursively delete the mod
+                        Directory.Delete($@"{this.BannerlordPath.Text}\Modules\Coop", true);
+                    }catch(Exception)
+                    {
+                        //renable the UI for other controls and notify that mod cannot be fully deleted
+                        this.Status.Text = "Ensure all files and processes are closed before reinstalling.";
+                        this.BannerlordPath.Enabled = true;
+                        this.ShortcutCheckbox.Enabled = true;
+                        this.PathBrowse.Enabled = true;
+                        this.Install.Enabled = true;
+                        return;
+                    }
+                    
+                }
+                else if (dialogResult == DialogResult.No)
+                {
+                    // renable the UI and allow changes to selections, return control
+                    this.BannerlordPath.Enabled = true;
+                    this.ShortcutCheckbox.Enabled = true;
+                    this.PathBrowse.Enabled = true;
+                    this.Install.Enabled = true;
+                    return;
+                }
+            }
+
+            // perform installation
+            PerformInstall();
+
+
+
+
+
+            // creating a shortcut
+            if (this.ShortcutCheckbox.Checked)
+            {
+                IShellLink link = (IShellLink)new ShellLink();
```

Comment:
do we need to coerce this here? Is a ShellLink not an IShellLink? Seems silly

Reply:
Might be a cleaner way. Will investigate.